src/compiler/compiler.rs

```diff
@@ -151,25 +173,31 @@ impl<F: Field + Hash> Compiler<F> {
                 self.add_step_type_handlers(ctx, symbols, machine_id);
 
                 for state_id in machine.keys() {
-                    ctx.step_type_def(self.get_step_type_handler(machine_id, state_id), |ctx| {
-                        self.add_internals(ctx, symbols, machine_id, state_id);
-
-                        ctx.setup(|ctx| {
-                            let pis = self.translate_queries(symbols, setup, machine_id, state_id);
-                            pis.iter().for_each(|pi| {
-                                let constraint = Constraint {
-                                    annotation: format!("{:?}", pi),
-                                    expr: pi.clone(),
-                                    typing: Typing::AntiBooly,
-                                };
-                                ctx.constr(constraint);
+                    ctx.step_type_def(
+                        self.mapping.get_step_type_handler(machine_id, state_id),
+                        |ctx| {
+                            self.add_internals(ctx, symbols, machine_id, state_id);
+
+                            ctx.setup(|ctx| {
+                                let pis =
+                                    self.translate_queries(symbols, setup, machine_id, state_id);
+                                pis.iter().for_each(|pi| {
```

Comment:
What does pi stand for?

Reply:
polynomial identity, like constraint, it is true that it is a bit obscure

Reply:
Let's rename it then :)

Reply:
to what do you think we should rename it?

Reply:
poly_identity